src/java.desktop/share/classes/javax/swing/text/DefaultCaret.java

```diff
@@ -366,6 +366,8 @@ protected void moveCaret(MouseEvent e) {
         }
     }
 
+    private int savedBlinkRate = 0;
+    private boolean blinkRateSaved = false;
```

Comment:
Naming of `savedBlinkRate` and `blinkRateSaved` looks almost the same and confusing for me.
Probably `isBlinkRateSaved` would be better to highlight boolean variable.

Reply:
Fixed.